Source/AGXUnreal/Private/Terrain/AGX_MovableTerrainComponent.cpp

```diff
@@ -0,0 +1,923 @@
+#include "Terrain/AGX_MovableTerrainComponent.h"
+#include "Terrain/AGX_ShovelComponent.h"
+#include "Shapes/HeightFieldShapeBarrier.h"
+#include "Shapes/AGX_ShapeComponent.h"
+#include "Materials/AGX_ShapeMaterial.h"
+#include "Materials/AGX_TerrainMaterial.h"
+#include "Utilities/AGX_StringUtilities.h"
+#include "Utilities/AGX_NotificationUtilities.h"
+#include "AGX_InternalDelegateAccessor.h"
+#include "AGX_RigidBodyComponent.h"
+#include "AGX_LogCategory.h"
+#include "AGX_Simulation.h"
+#include "TimerManager.h"
+#include "Engine/SimpleConstructionScript.h"
+#include "Engine/SCS_Node.h"
+#include "NiagaraComponent.h"
+#include "NiagaraDataInterfaceArrayFunctionLibrary.h"
+#include "NiagaraFunctionLibrary.h"
+
+UAGX_MovableTerrainComponent::UAGX_MovableTerrainComponent(
+	const FObjectInitializer& ObjectInitializer)
+	: UProceduralMeshComponent(ObjectInitializer)
+{
+	PrimaryComponentTick.bCanEverTick = true;
+	SetCanEverAffectNavigation(false);
+}
+
+void UAGX_MovableTerrainComponent::BeginPlay()
+{
+	Super::BeginPlay();
+
+	// Create Heights
+	InitializeHeights();
+
+	// Create Native
+	CreateNative();
+	
+	// Create Mesh(s) and Tile(s)
+	InitializeMesh();
+
+	// Update Mesh each tick
+	if (UAGX_Simulation* Simulation = UAGX_Simulation::GetFrom(this))
+	{
+		PostStepForwardHandle =
+			FAGX_InternalDelegateAccessor::GetOnPostStepForwardInternal(*Simulation)
+				.AddLambda(
+					[this](double)
+					{
+						//Update CurrentHeights
+						NativeBarrier.GetHeights(CurrentHeights, true);
+
+						//Rebuild mesh if needed
+						auto DirtyHeights = NativeBarrier.GetModifiedVertices();
+						if (DirtyHeights.Num() > 0)
+							UpdateMesh(DirtyHeights);
+					});
+	}
+
+	// Init particles
+	if (bEnableParticleRendering)
+	{
+		if (!InitializeParticles())
+		{
+			UE_LOG(
+				LogAGX, Warning,
+				TEXT("InitializeParticles returned false in AGX_MovableTerrainComponent '%s'. "
+					 "Ensure the selected Niagara System is valid."),
+				*GetName());
```

Comment:
To make life easier for our users: I suggest going through all warnings/error logs in this class and see if NotificationUtilities can be used instead. That way the User gets feedback clearly and is not missed if the user does'nt have the log open.

The notification utilities Notification will look like this: 

![image](https://github.com/user-attachments/assets/7894d5c4-fa53-49ce-a73f-15c28a7f90c0)


For things that can be printed out several times, the Log is probalby better than Notification. Notifications should only be used for stuff  that's typically triggered at most once. Otherwise they stack on top of each other and look bad.